src/smart-components/workspaces/WorkspaceListTable.tsx

```diff
@@ -107,6 +118,47 @@ const WorkspaceListTable = () => {
     setIsDeleteModalOpen(!isDeleteModalOpen);
   };
 
+  const isValidEditType = (workspace: Workspace) => {
+    switch (workspace.type) {
+      case 'root':
+        return false;
+      case 'default':
+        return true;
+      case 'ungrouped-hosts':
+        return false;
+      case 'standard':
+        return true;
+    }
+  };
+
+  const isValidDeleteType = (workspace: Workspace) => {
+    switch (workspace.type) {
+      case 'root':
+        return false;
+      case 'default':
+        return false;
+      case 'ungrouped-hosts':
+        return false;
+      case 'standard':
+        return true;
+    }
+  };
```

Comment:
Let's move these two functions out of the component so we don't have to re-run their calculataion on each render.